### PR TITLE
consider alternative summations in checkSummations

### PR DESCRIPTION
**Pull request by fbenke-pik**
This removes a bug with alternative summation rules, e.g. two alternative summations for `Final Energy|Industry` (annotated as `Final Energy|Industry` and `Final Energy|Industry 2` in the summations file). 
Before, the summation applies the first summation twice and checks if it adds up, ignoring the second summation rule.

### CODE REVIEW

File: R/checkSummations.R [hunk 1]
```diff
@@ -29,49 +30,68 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
...
+  message("The filtered data contains ", length(unique(data$variable)), " variables.")
+
+  if (nrow(data) == 0) {
+    return(NULL)
   }
+  tmp <- NULL
+
+  # iterate over summation rules
+  for (i in seq(1, length(checkVariables))) {
+    parentVar <- gsub(" [1-9]$", "", names(checkVariables[i]))
```
Comment (orichters) [inline R/checkSummations.R:61]: is that not just `parentVariables[i]`?

File: R/checkSummations.R [hunk 2]
```diff
@@ -29,49 +30,68 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
...
+    filter(!!sym("variable") %in% unique(c(parentVariables, unlist(checkVariables, use.names = FALSE))))
+  message("The filtered data contains ", length(unique(data$variable)), " variables.")
+
+  if (nrow(data) == 0) {
+    return(NULL)
   }
+  tmp <- NULL
+
+  # iterate over summation rules
+  for (i in seq(1, length(checkVariables))) {
```
Comment (orichters) [inline R/checkSummations.R:60]: you could use `seq_along(checkVariables)` instead

File: R/checkSummations.R [hunk 3]
```diff
@@ -29,49 +30,68 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
...
+
+    # create comparison for summation rule
+    parent <- filter(data, !!sym("variable") == parentVar) %>%
+      mutate(variable = names(checkVariables[i]))
+    children <- filter(data, !!sym("variable") %in% checkVariables[[i]]) %>%
+      rename("child" = !!sym("variable"), "childVal" = !!sym("value"))
+    comp <- left_join(parent, children, by = c("model", "scenario", "region", "unit", "period"))
+
+    # for summation groups the factor column can be used
+    if (exists("summationGroups")) {
```
Comment (orichters) [inline R/checkSummations.R:74]: how can `summationGroups` not exist here? Shouldn't the question be whether a `factor` column exists?

File: R/checkSummations.R [hunk 4]
```diff
@@ -29,49 +30,68 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
...
+        diff = !!sym("checkSum") - !!sym("value"),
+        reldiff = 100 * (!!sym("checkSum") - !!sym("value")) / !!sym("value")
+      )
+
+    tmp <- rbind(tmp, comp)
+  }
 
   # write data to dataDumpFile
-  if (length(dataDumpFile) > 0) {
+  if (!is.null(outputDirectory) && length(dataDumpFile) > 0) {
```
Comment (orichters) [inline R/checkSummations.R:94]: thanks for catching that!

**Review by orichters — APPROVED** [review]

File: R/checkSummations.R [hunk 5]
```diff
@@ -29,49 +30,68 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
...
+    # create comparison for summation rule
+    parent <- filter(data, !!sym("variable") == parentVar) %>%
+      mutate(variable = names(checkVariables[i]))
+    children <- filter(data, !!sym("variable") %in% checkVariables[[i]]) %>%
+      rename("child" = !!sym("variable"), "childVal" = !!sym("value"))
+    comp <- left_join(parent, children, by = c("model", "scenario", "region", "unit", "period"))
+
+    # for summation groups the factor column can be used
+    if (exists("summationGroups")) {
+      comp <- left_join(comp, select(summationGroups, c("child", "factor")), by = c("child"))
```
Comment (orichters) [inline R/checkSummations.R:75]:
```suggestion
      comp <- left_join(comp, select(summationGroups, c("child", "factor")), by = c("child"), relationship = "many-to-many")
```
to fix @Renato-Rodrigues problem: https://github.com/pik-piam/piamInterfaces/pull/143

File: R/checkSummations.R (hunk 3 above)
Comment (fbenke-pik) [reply R/checkSummations.R:74]: This is a check from the future that will become relevant once this is also used to check summations as part of the REMIND reporting. I will remove it for now and add it back again in a future PR.

**Pull request merged**